### PR TITLE
[Infra] `ti test` now supports `-t/--threads` for specifying number of testing threads

### PR DESCRIPTION
**Pull request by yuanming-hu**
Related issue = #601 

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

### CODE REVIEW

File: python/taichi/main.py [hunk 1]
```diff
@@ -17,46 +19,52 @@ def test_python(test_files=(), verbose=False):
...
-            args.append(os.path.join(test_dir, f))
+            pytest_args.append(os.path.join(test_dir, f))
     else:
         # run all the tests
-        args = [test_dir]
+        pytest_args = [test_dir]
     if verbose:
-        args += ['-s', '-v']
+        pytest_args += ['-s', '-v']
     if len(test_files) == 0 or len(test_files) > 4:
```
Comment (archibate) [inline python/taichi/main.py]: Seems -t and TI_TEST_THREADS only work when test_files > 4?

File: python/taichi/main.py [hunk 2]
```diff
@@ -17,46 +19,52 @@ def test_python(test_files=(), verbose=False):
...
                     [os.path.join(root_dir, 'misc/empty_pytest.py'),
-                     '-n1'])) == 0:  # if pytest has xdist
+                     '-n1'])) == 0:  # test if pytest has xdist or not
             try:
                 from multiprocessing import cpu_count
                 threads = min(8,
                               cpu_count())  # To prevent running out of memory
             except:
                 threads = 2
+            arg_threads = None
```
Comment (archibate) [inline python/taichi/main.py]: Reduce 1 indent from here to make -t works even if one file.

File: .travis.yml [hunk 3]
```diff
@@ -56,7 +56,7 @@ script:
 - export TAICHI_REPO_DIR=$TRAVIS_BUILD_DIR
 - export PYTHONPATH=$TAICHI_REPO_DIR/python
 - export PATH=$TAICHI_REPO_DIR/bin:$PATH
-- ti test -v && cd python && $PYTHON build.py try_upload
+- ti test -t 2 -v && cd python && $PYTHON build.py try_upload
```
Comment (archibate) [inline .travis.yml]: Maybe we should let multiprocessing.cpu_count to auto-detect how much cores CI have? Or you may want to set `threads = 2` for `cpucount == 1`?

Comment (yuanming-hu) [reply .travis.yml]: `multiprocessing.cpu_count` sometimes crashes on the buildbots, unfortunately.

> Or you may want to set `threads = 2` for `cpucount == 1`?

You mean hyperthreading?

Comment (archibate) [reply .travis.yml]: Sorry to hear that, may due to VMs, can keep this for now.
Yes, hyperthreading cpus want this to ultilize their resources.

Comment (yuanming-hu) [reply .travis.yml]: Sounds good.

**Comment by yuanming-hu** [comment]
Reverting back to 2 threads since using 4 threads seems unstable and does not improve testing speed empirically.

**Pull request merged**